Remotion/Web/Core/WebString.cs

```diff
@@ -0,0 +1,242 @@
+    }
+
+    [CanBeNull]
+    private readonly string _value;
+    private readonly WebStringType _type;
+
+    /// <summary>
+    /// Gets the type of string represented by this <see cref="WebString"/>.
+    /// </summary>
+    /// <value>The type of string represented by this <see cref="WebString"/>.</value>
+    public WebStringType Type => _type;
+
+    /// <summary>
+    /// Tests whether the <see cref="WebString"/>s value is <see langword="null" /> or an empty string.
+    /// </summary>
+    public bool IsEmpty => string.IsNullOrEmpty (_value);
+
+    private WebString ([CanBeNull] string value, WebStringType type)
+    {
+      _value = value;
+      _type = type;
+    }
+
+    /// <summary>
+    /// Gets the raw string value of the <see cref="WebString"/> or an empty string in case the value is <see langword="null" />.
+    /// </summary>
+    [NotNull]
+    public string GetValue ()
+    {
+      return _value ?? string.Empty;
+    }
+
+    /// <summary>
+    /// Writes the <see cref="WebString"/>'s value to the specified <paramref name="writer"/>.
+    /// The value is escaped if necessary, depending on the <see cref="Type"/> of the <see cref="WebString"/>.
+    /// </summary>
+    /// <param name="writer">The <see cref="HtmlTextWriter"/> where the value will be appended to. Must not be <see langword="null" />.</param>
+    /// <exception cref="InvalidOperationException">The <see cref="WebString"/>'s type is not compatible with the operation.</exception>
+    public void Write ([NotNull] HtmlTextWriter writer)
+    {
+      ArgumentUtility.CheckNotNull (nameof (writer), writer);
+
+      switch (_type)
+      {
+        case WebStringType.Unencoded:
+          var value = GetValue();
+          var parts = value.Split (s_newlineSplitSeparators, StringSplitOptions.None);
+
+          for (var i = 0; i < parts.Length; i++)
+          {
+            HttpUtility.HtmlEncode (parts[i], writer);
+            if (i < parts.Length - 1)
+              writer.WriteBreak();
+          }
+
+          break;
+        case WebStringType.Encoded:
+          writer.Write (GetValue());
+          break;
+        default:
+          throw new ArgumentOutOfRangeException();
+      }
+    }
+
+    /// <summary>
+    /// Adds the specified markup <paramref name="attribute"/> to the <paramref name="writer"/> using the <see cref="WebString"/> value.
+    /// The value is escaped if necessary, depending on the <see cref="Type"/> of the <see cref="WebString"/>.
+    /// </summary>
+    /// <param name="writer">The <see cref="HtmlTextWriter"/> where the attribute will be added. Must not be <see langword="null" />.</param>
+    /// <param name="attribute">The attribute that is to be added.</param>
+    /// <exception cref="InvalidOperationException">The <see cref="WebString"/>'s type is not compatible with the operation.</exception>
+    public void AddAttribute ([NotNull] HtmlTextWriter writer, HtmlTextWriterAttribute attribute)
```

Comment:
Need to documetn what happens when WebString is empty. in the future, we might want to circle back and sckip empty attributes but this is out-of-scope right now because too many moving parts.

Reply:
Empty strings will render an attribute with an empty value. Since we convert `null` to an empty string we always render an empty value.

Reply:
yeah, I checked now in the code:
null makes just the attribute-name.
empty makes attribute-name + `=""`

Either way, no need to support null and the decision to just not render a blank attribute will be made later on.

There's also the annoying bit about when an attribute without value is even valid html. (html yes, xhtml no). It's best to keep semantics simple.